pkgs/top-level/python-packages.nix

```diff
@@ -21672,8 +21682,12 @@ in {
       export LANG="en_US.UTF-8";
     '';
 
-    patchPhase = ''
-      sed -i "s|_lgeos = load_dll('geos_c', fallbacks=.*)|_lgeos = load_dll('geos_c', fallbacks=['${pkgs.geos}/lib/libgeos_c.so'])|" shapely/geos.py
+    patchPhase = let
+      libgeos = if stdenv.isDarwin then "libgeos_c.dylib" else "libgeos_c.so";
```

Comment:
see `sharedLibraryExtension` in this file

Reply:
I initially wanted to do it like that. However I realized that it can only be applied to the `libgeos_c` case, since for `libc` we actually need to link against the `libc.so.6` version, as `libc.so` is a linker ASCII text file which `load_dll` cannot deal with.
So I decided to be consistent and bind both `libgeos` and `libc`.
If you prefer to use `sharedLibraryExtension` in the libgeos-case, I can change it to that.

Reply:
@FRidh: I incorporated your suggestion to use `sharedLibraryExtension` for libgeos_c